### ISSUE TITLE
Fix travis tests 

### ISSUE THREAD

**Issue by jdknives** [skycoin/skywire-testnet#210]
@evanlinjin write: This is a table of dodgy tests that have been encountered.

| Status | Test | Travis | Notes |
| --- | --- | --- | --- |
| ~ | `TestTransportMangerReEstablishTransports`<br>`/pkg/transport/manager_test.go:130` | https://travis-ci.com/watercompany/skywire/jobs/178729323 | |
| ~ | `TestClientConnectInitialServers` <br> `/pkg/messaging/client_test.go:51` | https://travis-ci.com/watercompany/skywire/jobs/179175588 | Handshake timeout again. Probably need to find a way of removing `time.Sleep()` calls in tests. |
| ~ | `TestChannelServeSocket` <br> `/internal/therealssh/channel_test.go:124` | https://travis-ci.com/watercompany/skywire/jobs/179183084 | |

**Comment by evanlinjin** [comment]
## TODO: Add the following.
Test: https://travis-ci.org/skycoin/skywire/jobs/504369100
```
==================
WARNING: DATA RACE
Write at 0x00c0000d8f10 by goroutine 25:
  runtime.closechan()
      /home/travis/.gimme/versions/go1.12.linux.amd64/src/runtime/chan.go:334 +0x0
  github.com/skycoin/skywire/pkg/messaging.(*channel).close()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/channel.go:148 +0x104
  github.com/skycoin/skywire/pkg/messaging.(*channel).Close()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/channel.go:122 +0x264
  github.com/skycoin/skywire/internal/ioutil.(*AckReadWriter).Close()
      /home/travis/gopath/src/github.com/skycoin/skywire/internal/ioutil/ack.go:124 +0x8d
  github.com/skycoin/skywire/pkg/messaging.(*ackedChannel).Close()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/channel.go:220 +0x6c
  github.com/skycoin/skywire/pkg/messaging.TestClientDial()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/client_test.go:118 +0x13bd
  testing.tRunner()
      /home/travis/.gimme/versions/go1.12.linux.amd64/src/testing/testing.go:865 +0x163
Previous read at 0x00c0000d8f10 by goroutine 54:
  runtime.chansend()
      /home/travis/.gimme/versions/go1.12.linux.amd64/src/runtime/chan.go:142 +0x0
  github.com/skycoin/skywire/pkg/messaging.(*Client).onData.func1()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/client.go:322 +0x1ca
Goroutine 25 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.12.linux.amd64/src/testing/testing.go:916 +0x699
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.12.linux.amd64/src/testing/testing.go:1157 +0xa8
  testing.tRunner()
      /home/travis/.gimme/versions/go1.12.linux.amd64/src/testing/testing.go:865 +0x163
  testing.runTests()
      /home/travis/.gimme/versions/go1.12.linux.amd64/src/testing/testing.go:1155 +0x523
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.12.linux.amd64/src/testing/testing.go:1072 +0x2eb
  github.com/skycoin/skywire/pkg/messaging.TestMain()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/client_test.go:24 +0xae
  main.main()
      _testmain.go:126 +0x330
Goroutine 54 (finished) created at:
  github.com/skycoin/skywire/pkg/messaging.(*Client).onData()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/client.go:318 +0x1280
  github.com/skycoin/skywire/pkg/messaging.(*Client).onData-fm()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/client.go:264 +0x79
  github.com/skycoin/skywire/pkg/messaging.(*Pool).messageAction.func2()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/pool.go:215 +0x96
  github.com/skycoin/skywire/pkg/messaging.(*Link).readFrame()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/link.go:186 +0xb8
  github.com/skycoin/skywire/pkg/messaging.(*Link).handleData()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/link.go:169 +0x1c0
  github.com/skycoin/skywire/pkg/messaging.(*Link).readLoop()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/link.go:152 +0x110
  github.com/skycoin/skywire/pkg/messaging.(*Link).Open.func1()
      /home/travis/gopath/src/github.com/skycoin/skywire/pkg/messaging/link.go:85 +0x54
==================
--- FAIL: TestClientDial (3.29s)
    testing.go:809: race detected during execution of test
FAIL
coverage: 81.7% of statements
FAIL	github.com/skycoin/skywire/pkg/messaging	21.077s
```

**Comment by ayuryshev** [comment]
It looks like `make lint` locally and in travis-ci works different.

Code like:
```golang
var pkCmd = &cobra.Command{
	Use:   "pk",
	Short: "get public key of node",
	Run: func(_ *cobra.Command, args []string) { //unused args

		client := rpcClient()
		summary, err := client.Summary()
		if err != nil {
			log.Fatal("Failed to connect:", err)
		}

		fmt.Println(summary.PubKey)
	},
}
``` 
checked with `make lint` without problems in development environment
and failed in travis-ci

**Comment by evanlinjin** [comment]
> It looks like `make lint` locally and in travis-ci works different.
> 
> Code like:
> 
> ```go
> var pkCmd = &cobra.Command{
> 	Use:   "pk",
> 	Short: "get public key of node",
> 	Run: func(_ *cobra.Command, args []string) { //unused args
> 
> 		client := rpcClient()
> 		summary, err := client.Summary()
> 		if err != nil {
> 			log.Fatal("Failed to connect:", err)
> 		}
> 
> 		fmt.Println(summary.PubKey)
> 	},
> }
> ```
> 
> checked with `make lint` without problems in development environment
> and failed in travis-ci

Good spotting. Please fix!

**Comment by gz-c** [comment]
make sure a pinned version of gometalinter is used by Travis. refer to skycoin's ci-scripts for installing the pinned version in Travis

**Comment by jdknives** [comment]
Closed by #244
